package/src/renderer/Canvas.tsx

```diff
@@ -31,29 +32,47 @@ export interface CanvasProps extends SkiaBaseViewProps {
   onTouch?: TouchHandler;
 }
 
+const useOnSizeEvent = (resultValue: SkiaBaseViewProps["onSize"]) => {
+  const onSize = useValue({
+    width: 0,
+    height: 0,
+  });
+
+  useLayoutEffect(() => {
+    if (!resultValue) {
+      return;
+    }
+
+    return onSize.addListener((newValue) => {
+      const currentValue = isValue(resultValue)
+        ? resultValue.current
+        : resultValue.value;
+
+      if (
```

Comment:
This if statement is probably unnecessary since we appear to check for it on the C++ side but we will double check.